plugins/plugin-kiali/src/components/panel/Graph.tsx

```diff
@@ -232,71 +230,84 @@ const Graph: React.FunctionComponent<IGraphProps> = ({ instance, times, edges, n
+      layout.current.run();
+    }
+  }, [edges, nodes, size]);
 
-  // useEffect is used to adjust the size of the topology graph container, so it takes the maximum available space.
   useEffect(() => {
-    if (containerRef && containerRef.current) {
-      setWidth(containerRef.current.getBoundingClientRect().width);
-      setHeight(containerRef.current.getBoundingClientRect().height);
+    if (!container.current) {
+      return;
     }
-  }, [edges, nodes]);
 
-  // useEffect is used to remove the tab listener, when the component is unmounted.
+    try {
+      if (!graph.current) {
+        cytoscape.use(dagre);
+        nodeHtmlLabel(cytoscape);
+
+        graph.current = cytoscape({
+          container: container.current,
+          elements: {
+            edges: edges as cytoscape.EdgeDefinition[],
+            nodes: nodes as cytoscape.NodeDefinition[],
+          },
+          style: styleSheet,
+        });
+        graph.current.on('tap', onTap);
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (graph.current as any).nodeHtmlLabel([
+          {
+            halign: 'center',
+            halignBox: 'center',
+            query: 'node:visible',
+            tpl: nodeLabel,
+            valign: 'bottom',
+            valignBox: 'bottom',
+          },
+        ]);
+      }
+    } catch (err) {}
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [size]);
+
   useEffect(() => {
     return (): void => {
-      if (graphRef.current) {
-        graphRef.current.removeListener('tab', onTap);
-        graphRef.current = null;
+      if (graph.current) {
+        graph.current.removeListener('tab', onTap);
+        graph.current.destroy();
       }
     };
-  }, [onTap]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   return (
-    <div style={{ minHeight: '100%' }} ref={containerRef}>
-      {height > 0 && width > 0 ? (
-        <CytoscapeComponent
-          elements={CytoscapeComponent.normalizeElements({ edges: edges, nodes: nodes })}
-          style={{ height: `${height}px`, width: `${width}px` }}
-          zoomingEnabled={true}
-          maxZoom={5}
-          minZoom={0.1}
-          autounselectify={false}
-          boxSelectionEnabled={true}
-          layout={layout}
-          stylesheet={styleSheet}
-          cy={cyCallback}
-        />
+    <div style={{ minHeight: '100%' }} ref={wrapper}>
+      {size.height > 0 && size.width > 0 ? (
+        <div style={{ height: `${size.height}px`, width: `${size.width}px` }} ref={container} />
       ) : null}
     </div>
   );
 };
 
-export default memo(Graph, (prevProps, nextProps) => {
-  if (
-    JSON.stringify(prevProps.edges) === JSON.stringify(nextProps.edges) &&
-    JSON.stringify(prevProps.nodes) === JSON.stringify(nextProps.nodes)
-  ) {
-    return true;
-  }
+export default Graph;
+// export default memo(Graph, (prevProps, nextProps) => {
```

Comment:
Can this be removed or do you want to keep this for later?

Reply:
Good catch 👍